bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -199,6 +200,28 @@ public Response update(@PathParam("thingUID") String thingUID, ThingBean thingBe
 
         return Response.ok().build();
     }
+    
+    @PUT
+    @Path("/{thingUID}/action/{actionId}")
+    @Consumes(MediaType.TEXT_PLAIN)
+    public Response action(@PathParam("thingUID") String thingUID, @PathParam("actionId") String actionId,
+            String actionValue) {
+
+        Thing thing = thingRegistry.get(new ThingUID(thingUID));
+        if (thing == null) {
+            logger.warn("Received HTTP PUT request at '{}' for the unknown thing '{}'.", uriInfo.getPath(), thingUID);
+            return Response.status(Status.NOT_FOUND).build();
+        }
+
+        if(thing instanceof ThingConfigAction) {
```

Comment:
'Would prefer a non-invasive design w/o touching core.thing at all. 'Thinking of a service for config-action to ask for, and trigger, iff applicable.